docs/data/joy/components/checkbox/HoverCheckbox.tsx

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+import Checkbox from '@mui/joy/Checkbox';
+import Done from '@mui/icons-material/Done';
+
+export default function HoverCheckbox() {
+  return (
+    <Checkbox
+      uncheckedIcon={<Done />}
+      label="My unchecked icon appears on hover"
+      slotProps={{
+        root: ({ checked, focusVisible }) => ({
+          sx: !checked
+            ? {
+                '& svg': { opacity: focusVisible ? 0.32 : 0 },
+                '&:hover svg': {
+                  opacity: 0.32,
+                },
+              }
+            : undefined,
```

Comment:
was getting below typescript error if `sx` is `{}` hence changed to `undefined`. Not entirely sure why it worked though
<img width="726" alt="Screenshot 2023-03-01 at 6 06 03 PM" src="https://user-images.githubusercontent.com/60743144/222144138-b7632a27-a344-459d-a312-234977b9e5c7.png">